### PR TITLE
Add Login page functionality

### LINKED ISSUE #24: Login page functionality

### PR DESCRIPTION
**Pull request by flacial**
### Changes

- Export the socket instance to its own file
- Create global types
- Fix an issue with eslint that causes ts syntax to be unrecognized by extending babel-eslint
- Clean up the App page by removing unused code

Closes #24

### CODE REVIEW

File: client/src/view/Login.svelte [hunk 1]
```diff
@@ -14,7 +34,9 @@
     <span>proximity</span>
   </div>
   <button
-    class="button orange-gradient px-12 py-3 font-medium transition-shadow hover:shadow-2xl hover:shadow-orange-600/50 focus:outline-none focus:ring focus:ring-orange-300 active:opacity-90"
+    class="button orange-gradient px-12 py-3 font-medium transition-shadow hover:shadow-2xl hover:shadow-orange-600/50 focus:outline-none focus:ring focus:ring-orange-300 active:opacity-90 disabled:opacity-40 disabled:shadow-none"
+    on:click={handleSignIn}
+    disabled={buttonDisabled}
```
Comment (JasirZaeem) [inline client/src/view/Login.svelte:39]: Some indication that a background blocking process is happening in addition to disabling the button will be helpful. Some animation and/or text like "Getting location...", "Logging in..." etc.

File: client/src/type/global.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export type User = {
+  hash?: string;
+  username?: string;
+};
+
+export type Users = string[];
```
Comment (JasirZaeem) [inline client/src/type/global.ts]: Can make this name more specific, `Users` on a glance would imply `User[]`, if you are storing usernames in this you can call it Usernames, or UserHashes if you storing hashes.

**Review by JasirZaeem — APPROVED** [review]

File: client/src/type/global.ts (hunk 2 above)
Comment (flacial) [reply client/src/type/global.ts]: Yeah, `UserHashes` sounds suitable.

File: client/src/view/Login.svelte (hunk 1 above)
Comment (flacial) [reply client/src/view/Login.svelte:39]: I agree. I'm planning to add these later.

**Pull request merged**